test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,314 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
+                  type: "number",
+                  enum: [1, 2, 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(0)
+    })
+  })
+
+  //Array Tests
+
+  it("should return an error for a non array value in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [1, 2, 3]                  
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const secondError = allErrors[1]
+      expect(allErrors.length).toEqual(2)
+      expect(secondError.path).toEqual(["paths", "/pets", "get", "parameters", "0", "schema", "enum", 2])
+      expect(secondError.level).toEqual("warning")
+      expect(secondError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+  
+  it("should not return a type error when all items are array in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [[1,2],[3,4]]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      // It's throwing another error
+      expect(allErrors.length).toEqual(1)
+    })
+  })
+
+  //Object Tests
+
+  it("should return an error for a non object value (array) in a enum object type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "objectSample",
+                in: "query",
+                schema: {
+                  type: "object",
+                  enum: [[1,3], 2, 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const firstError = allErrors[0]
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
```

Comment:
**Change request**: add an expectation for the path of the error as well

Reply:
Expectation for path added, and Include pattern implemented.